Sources/SwiftParser/Parser+Entry.swift.gyb

```diff
@@ -0,0 +1,75 @@
+%{
+  from gyb_syntax_support import *
+  # -*- mode: Swift -*-
+  # Ignore the following admonition it applies to the resulting .swift file only
+}%
+//// Automatically Generated From Entry.swift.gyb.
+//// Do Not Edit Directly!
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2022 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+@_spi(RawSyntax) import SwiftSyntax
+
+extension Parser {
+  /// Parse the source code in the given string as Swift source file. See
+  /// `Parser.init` for more details.
+  public static func parse(
+    source: String,
+    parseTransition: IncrementalParseTransition? = nil
+  ) -> SourceFileSyntax {
+    var parser = Parser(source)
+    return SourceFileSyntax.parse(from: &parser)
+  }
+
+  /// Parse the source code in the given string as Swift source file. See
+  /// `Parser.init` for more details.
+  public static func parse(
+    source: UnsafeBufferPointer<UInt8>,
+    maximumNestingLevel: Int? = nil,
+    parseTransition: IncrementalParseTransition? = nil
+  ) -> SourceFileSyntax {
+    var parser = Parser(source, maximumNestingLevel: maximumNestingLevel)
+    return SourceFileSyntax.parse(from: &parser)
+  }
+}
+
+public protocol SyntaxParseable: SyntaxProtocol {
+  static func parse(from parser: inout Parser) -> Self
+}
+
+% for node in SYNTAX_NODES:
+%   if node.parser_function:
+extension ${node.name}: SyntaxParseable {
+  public static func parse(from parser: inout Parser) -> ${node.name} {
+    let node = parser.${node.parser_function}()
+    return parser.parseRemainder(into: node).syntax
```

Comment:
Are we 100% sure that `parser` wouldn't be deinitialized before calling `syntax`?

Reply:
It would seem super broken to me if it could since it's passed in as an `inout`. I suppose theoretically we could know that it's not used after this call + the last use is on the `parseRemainder`, but... 🤷 I'm happy to add a `withExtendedLifetime` if you think it needs it.

Reply:
My concern was when this function is inlined within the module, the optimizer might decide to release it early. But as far as I tried, it doesn't seem to be the case. So let's keep this as is.